### PR TITLE
automated tests & code documentation

### PR DESCRIPTION
**Pull request by derisen**
what to do next: 
-increase code coverage (curr. %50)
-expand documentation
-practice code refactoring

### CODE REVIEW

File: src/main/java/ca/nines/wilde/Main.java [hunk 1]
```diff
@@ -14,22 +14,37 @@
...
         if (args.length > 0) {
             commandName = args[0];
         }
-        if (!commandList.containsKey(commandName)) {
-            System.err.println("Unknown command: " + commandName);
-            commandName = "help";
+        
+        if(!commandList.containsKey(commandName)) {
+                commandName = "help";
+                throw new IllegalArgumentException("Unknown command: " + commandName);
```
Comment (ubermichael) [inline src/main/java/ca/nines/wilde/Main.java]: Why did you change this to throw an exception? The user will only see the exception output and none of the help output.

File: src/main/java/ca/nines/wilde/Main.java [hunk 2]
```diff
@@ -40,9 +55,10 @@ public static void main(String[] args) throws InstantiationException, IllegalAcc
             System.err.println(e.getMessage());
             System.exit(-1);
         }
+        
         if (commandLine.hasOption("help")) {
             HelpFormatter formatter = new HelpFormatter();
-            formatter.printHelp("java -jar wip.jar " + cmd.getCommandName() + " [options]", opts);
+            formatter.printHelp("java -jar oscar.jar " + cmd.getCommandName() + " [options]", opts);
```
Comment (ubermichael) [inline src/main/java/ca/nines/wilde/Main.java]: Good catch!

File: src/main/java/ca/nines/wilde/Util/Text.java [hunk 3]
```diff
@@ -11,11 +11,19 @@
 import org.apache.commons.text.similarity.LevenshteinDistance;
 
 /**
- *
+ * This class contains methods to calculate the similarity distance of two strings,
+ * as well as utility methods like text normalizer.
```
Comment (ubermichael) [inline src/main/java/ca/nines/wilde/Util/Text.java]: Distance and similarity are related but different concepts. If two things have a large distance then they have very little similarity. This class calculates the similarity.

File: src/main/java/ca/nines/wilde/Util/Text.java [hunk 4]
```diff
@@ -29,14 +37,25 @@ public static double levenshtein(String a, String b) {
         }
         double similarity = 1.0 - (distance / ((double) maxLength));
         return similarity;
+        
     }
+    
+    /** This method implements the Cosine Distance algorithm.
```
Comment (ubermichael) [inline src/main/java/ca/nines/wilde/Util/Text.java]: This is the cosine similarity. It uses the cosine distance to calculate the cosine similiarity.

File: src/main/java/ca/nines/wilde/cmd/ParagraphCompare.java [hunk 5]
```diff
@@ -93,7 +94,7 @@ public boolean include(WildeDoc doc) throws XPathExpressionException {
             documentI.setParagraphsIndexed();
             writer.write(documentI.getPath(), documentI);
         }
-
+        System.out.println("\nDone");
```
Comment (ubermichael) [inline src/main/java/ca/nines/wilde/cmd/ParagraphCompare.java]: Is this necessary? I'd prefer code that exits cleanly.

File: src/main/java/ca/nines/wilde/doc/Cleaner.java [hunk 6]
```diff
@@ -35,7 +37,13 @@
         "dc.source.url",
         "wr.path",
     })));
-
+    
+    /**
+    * This method removes all the tags in an XML document that are not marked as KEPT above.
```
Comment (ubermichael) [inline src/main/java/ca/nines/wilde/doc/Cleaner.java]: It only removes the metadata tags. Other HTML tags are untouched.

File: src/main/java/ca/nines/wilde/doc/WildeDoc.java [hunk 7]
```diff
@@ -29,20 +30,35 @@
...
+    */
     public WildeDoc(Path path, Document document) {
         this.document = document;
         this.xpath = XPathFactory.newInstance().newXPath();
         this.path = path;
     }
-
+    
+    /** 
+    * @return a path expression
```
Comment (ubermichael) [inline src/main/java/ca/nines/wilde/doc/WildeDoc.java]: This is (or should be) the file system path to the original XML file that was used to create the object. Calling it a path expression will likely lead to confusion with xpath expressions.

**Comment by derisen** [comment]
Got it! I'll fix them and also do some code cleaning and then repush...
